common/src/main/java/com/scottlogic/deg/common/profile/Types.java

```diff
@@ -2,19 +2,20 @@
 
 import java.time.OffsetDateTime;
 import java.util.function.Function;
+import java.util.function.Predicate;
 
 public enum Types {
     NUMERIC(o -> o instanceof Number),
     STRING(o -> o instanceof String),
     DATETIME(o -> o instanceof OffsetDateTime);
 
-    private final Function<Object, Boolean> isInstanceOf;
+    private final Predicate<Object> isInstanceOf;
 
-    Types(final Function<Object, Boolean> isInstanceOf) {
+    Types(final Predicate<Object> isInstanceOf) {
         this.isInstanceOf = isInstanceOf;
     }
 
     public boolean isInstanceOf(Object o) {
-        return isInstanceOf.apply(o);
+        return isInstanceOf.equals(o);
```

Comment:
This doesn't seem correct. I imagine we want to check if the applied method `java.util.function.Predicate#test` for `instanceOf` returns true (ie. `o` is an instance of `T`), not check for equality on the predicate itself.